NBitcoin/Utils.cs

```diff
@@ -416,6 +416,32 @@ public static bool ArrayEqual(byte[] a, int startA, byte[] b, int startB, int le
 			return true;
 		}
 
+		public static bool NullSafeEquals<T>(T a, T b) where T : IEquatable<T> =>
+			(a == null && b == null) || ((a != null && b != null) && a.Equals(b));
+
+		public static bool DictEqual<T, U>(Dictionary<T, U> x, Dictionary<T, U> y)
+		{
+			if (null == y)
+				return null == x;
+			if (null == x)
+				return false;

```

Comment:
These four lines are not really necessary because you use `object.ReferenceEquals()` that does the same.